### PR TITLE
Migrate all assertions to AssertJ

### PR DESCRIPTION
**Pull request by 3flex**
Retires the "kotlin-test" dependency in favour of AssertJ, and gets rid of the single JUnit Jupiter assertion (also switched to AssertJ).

Addresses the assertions part of #1170.

### CODE REVIEW

File: detekt-cli/src/test/kotlin/io/gitlab/arturbosch/detekt/cli/console/BuildFailureReportSpec.kt [hunk 1]
```diff
@@ -41,23 +42,24 @@ internal class BuildFailureReportSpec : SubjectSpek<BuildFailureReport>({
...
+				assertThatExceptionOfType(BuildFailure::class.java).isThrownBy { subject.render(detektion) }
 			}
 
 			it("should contain no stacktrace on fail") {
 				subject.init(TestConfig(mapOf("maxIssues" to "-2")))
-				val exception = assertFails { subject.render(detektion) }
-				assertThat(exception.stackTrace).isEmpty()
+				assertThatExceptionOfType(BuildFailure::class.java)
+						.isThrownBy { subject.render(detektion) }
+						.withStackTraceContaining("io.gitlab.arturbosch.detekt.cli.console.BuildFailure: Build failed with 1 weighted issues (threshold defined was -2).")
```
Comment (3flex) [inline detekt-cli/src/test/kotlin/io/gitlab/arturbosch/detekt/cli/console/BuildFailureReportSpec.kt]: Should mention this as it's the only one I didn't see an obvious AssertJ method that allows testing that no stacktrace is shown, so the test is not exactly checking that the stacktrace is empty - `withStackTraceContaining` only checks the text exists in the stacktrace - it doesn't check that it's the only text.

Probably need a solution before merging.

Comment (Mauin) [reply detekt-cli/src/test/kotlin/io/gitlab/arturbosch/detekt/cli/console/BuildFailureReportSpec.kt]: We could still match it similar to the kotlin-test way.

```kotlin
try { 
  subject.render(detektion)
  fail("Render should throw")
} catch (e: Exception) {
  assertThat(e).isInstanceOf(BuildFailure::class.java)
  assertThat(e.stackTrace).isEmpty()
}
```

**Comment by 3flex** [comment]
Tweaked that test - should be good to merge.

**Comment by arturbosch** [comment]
+1 thanks!

**Pull request merged**